cupsfilters/ghostscript.c

```diff
@@ -92,13 +92,23 @@ parse_pdf_header_options(FILE *fp,
       char *p;
 
       p = strchr(buf + 19, ':');
+      if (!p)
+      {
+        h->NumCopies = 1;
+        continue;
+      }
       h->NumCopies = atoi(p + 1);
     }
     else if (strncmp(buf, "%%PDFTOPDFCollate", 17) == 0)
     {
       char *p;
 
       p = strchr(buf + 17, ':');
+      if (!p) {
+        h->Collate = CUPS_TRUE;
```

Comment:
Collating is IMO more complex processing (the job has to reorder the pages) and some devices do not support this, so I would go for CUPS_FALSE here.
The similar change in the other comments as well.